Products/Products-Root/Screens/Authentication/Splash/SplashViewController.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  SplashViewController.swift
+//  Products
+//
+//  Created by Dan Pop on 10/03/2020.
+//  Copyright © 2020 Archlime. All rights reserved.
+//
+
+import UIKit
+
+protocol SplashViewControllerDelegate: class {
+    func splashViewControlledDidPressSignIn()
+    func splashViewControllerDidPressRegister()
+}
+
+
+class SplashViewController: BaseViewController {
+    private var viewModel: SplashViewModel!
+    weak var delegate: SplashViewControllerDelegate?
```

Comment:
for readability, the ordering of properties must be inverted (public/internal first, private after)